fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1540,6 +1544,102 @@ void ReadYourWritesTransaction::getWriteConflicts( KeyRangeMap<bool> *result ) {
+				                             iter->value() ? LiteralStringRef("1") : LiteralStringRef("0")));
+			}
+		}
+	} else {
+		CoalescedKeyRefRangeMap<ValueRef> readConflicts{ LiteralStringRef("0"), specialKeys.end };
+		for (const auto& range : tr.readConflictRanges())
+			readConflicts.insert(range.withPrefix(readConflictRangeKeysRange.begin, result.arena()),
+			                     LiteralStringRef("1"));
+		for (const auto& f : tr.getExtraReadConflictRanges())
+			if (f.isReady() && f.get().first < f.get().second)
+				readConflicts.insert(KeyRangeRef(f.get().first, f.get().second)
+				                         .withPrefix(readConflictRangeKeysRange.begin, result.arena()),
+				                     LiteralStringRef("1"));
+		auto beginIter = readConflicts.rangeContaining(kr.begin);
+		if (beginIter->begin() != kr.begin) ++beginIter;
+		for (auto it = beginIter; it->begin() < kr.end; ++it) {
+			result.push_back(result.arena(), KeyValueRef(it->begin(), it->value()));
+		}
+	}
+	return result;
+}
+
+Standalone<RangeResultRef> ReadYourWritesTransaction::getWriteConflictRangeIntersecting(KeyRangeRef kr) {
+	if (writeConflictRangeUnknown) {
+		throw accessed_unreadable();
+	}
+	ASSERT(writeConflictRangeKeysRange.contains(kr));
+	Standalone<RangeResultRef> result;
+
+	if (!options.readYourWritesDisabled) {
+		kr = kr.removePrefix(writeConflictRangeKeysRange.begin);
+		WriteMap::iterator it(&writes);
+		it.skip(kr.begin);
+		if (it.beginKey() != allKeys.begin) --it;
+
+		bool inConflictRange = false;
+		ExtStringRef conflictBegin;
+
+		for (; it.beginKey() < kr.end; ++it) {
+			if (it.is_conflict_range() && !inConflictRange) {
+				conflictBegin = std::max(ExtStringRef(allKeys.begin), it.beginKey());
+				inConflictRange = true;
+			} else if (!it.is_conflict_range() && inConflictRange) {
+				KeyRangeRef keyrange(conflictBegin.toArena(result.arena()), it.beginKey().toArena(result.arena()));
+				if (kr.contains(keyrange.begin))
+					result.push_back(result.arena(), KeyValueRef(keyrange.begin.withPrefix(
+					                                                 writeConflictRangeKeysRange.begin, result.arena()),
+					                                             LiteralStringRef("1")));
+				if (kr.contains(keyrange.end))
+					result.push_back(result.arena(), KeyValueRef(keyrange.end.withPrefix(
+					                                                 writeConflictRangeKeysRange.begin, result.arena()),
+					                                             LiteralStringRef("0")));
+				inConflictRange = false;
+			}
+		}
+
+		if (inConflictRange) {
+			KeyRangeRef keyrange(conflictBegin.toArena(result.arena()), allKeys.end);
+			if (kr.contains(keyrange.begin))
+				result.push_back(result.arena(), KeyValueRef(keyrange.begin.withPrefix(
+				                                                 writeConflictRangeKeysRange.begin, result.arena()),
+				                                             LiteralStringRef("1")));
+			if (kr.contains(keyrange.end))
```

Comment:
Since `keyrange.end` is `allKeys.end`, which is `\xff\xff`, will this statement always be `false`?

Reply:
I think so, good catch.